### DISCUSSION TITLE
unxutils 2007.03.01  ---tail.exe: write error: No space left on device

### DISCUSSION THREAD

**Discussion by Archer456** [ScoopInstaller/Main#3302] [Q&A] [answered]
## Bug Report
unxutils 2007.03.01

**tail.exe** 

╰─ **$PSVersionTable.PSVersion**                                                                                                                                                       ─╯

Major  Minor  Patch  PreReleaseLabel BuildLabel
-----  -----  -----  --------------- ----------
7      2      1

╰─ scoop list                                                                                                                                                                      ─╯
Installed apps:

  7zip 21.07 [main]
  cacert 2022-02-01 [main]
  curl 7.81.0 [main]
  gcc 11.2.0 [main]
  grep 3.7 [main]
  jq 1.6 [main]
  neovim 0.6.1 [main]
  sudo 0.2020.01.26 [main]
  unxutils 2007.03.01 [main]
  wget 1.21.2 [main]

### Current Behaviour

![image](https://user-images.githubusercontent.com/18392901/153792519-75ae3073-6785-4d47-993e-0467fd99d4c3.png)

![image](https://user-images.githubusercontent.com/18392901/153792579-0b89d20e-fc2b-4af4-bbc0-94d71c5e9092.png)

![image](https://user-images.githubusercontent.com/18392901/153792893-038f4bf0-bbaf-4ae5-94eb-fd9f449719f3.png)

Problem 1: tail.exe: write error: No space left on device
Problem 2: Chinese display garbled characters

### Expected Behaviour

My C drive is confirmed to have enough disk space, why is the error "write error: No space left on device" reported?

I also confirmed that it has sufficient permissions, see screenshot above.

So, my desired result is that this command should output my logs normally.

### System details

Microsoft Windows [ 10.0.19044.1526]

64bit 

powershell 7.2.1

**Additional software:** [(optional) e.g. ConEmu, Git]

#### Scoop Configuration
{
  "lastupdate": "2022-02-14T08:59:15.9799741+08:00",
  "SCOOP_REPO": "https://github.com/ScoopInstaller/Scoop",
  "SCOOP_BRANCH": "master"
}

```json
//# Your configuration here
```

**Comment by rashil2000** [comment, marked as answer]
This looks like a problem with unxutils package, which is extremely extremely outdated. If you want more up to date coreutils, use the `coreutils`, `uutils-coreutils` or `busybox` (any one of these) package.

/cc @issaclin32

**Comment by Archer456** [comment]
> This looks like a problem with unxutils package, which is extremely extremely outdated. If you want more up to date coreutils, use the `coreutils`, `uutils-coreutils` or `busybox` (any one of these) package.
> 
> /cc @issaclin32

Thanks for your answer. Yes, I just moved from a linux environment to a windows environment and was looking for some linux-like command line toolkits.
Can you answer, which of these coreutils`, `uutils-coreutils` or `busybox` packages are more stable and better match the win environment?

**Comment by rashil2000** [comment]
I can't really comment on stability because I don't use any of them, but I'd recommend trying `coreutils` first.

`uutils-coreutils` is a rewrite of coreutils in Rust. See https://github.com/uutils/coreutils for more details.

`busybox` is a multi-call binary and supports common coreutils (Scoop will create shims for them).

**Reply by Archer456** [reply]
> I can't really comment on stability because I don't use any of them, but I'd recommend trying `coreutils` first.
> 
> `uutils-coreutils` is a rewrite of coreutils in Rust. See https://github.com/uutils/coreutils for more details.
> 
> `busybox` is a multi-call binary and supports common coreutils (Scoop will create shims for them).

![image](https://user-images.githubusercontent.com/18392901/153802509-b26dd014-3316-48a1-b053-f9dd35df3f98.png)
I tried to install coreutils, but the tail below cannot run normally under powershell.
I finally tried busybox, it works fine, I didn't encounter any major problems, I can run tail directly under powershell.

**Reply by rashil2000** [reply]
What error do you get with coreutils?

**Reply by Archer456** [reply]
> What error do you get with coreutils?

Prompt me, tail.exe cannot be found, not a cmdlet similar information

**Reply by rashil2000** [reply]
It works for me:

![image](https://user-images.githubusercontent.com/46838874/153807537-ca0258c7-3594-4926-9a0a-d4b3192a7084.png)

**Comment by rashil2000** [comment]
Or you can just use Bash (which comes with Git for Windows). It will have the most stable and latest unix tools.

**Comment by Archer456** [comment]
Yes, I have the windows version of git-bash. use it too.
Just want to find some similar linux tools under powershell.

**Comment by issaclin32** [comment]
Added **notes** that directs the user to use `uutils-coreutils` instead.
https://github.com/ScoopInstaller/Main/commit/dec7032a206ac6b1192c80a7e67028c52984a37f
